fastapi/exception_handlers.py

```diff
@@ -1,25 +1,34 @@
+from typing import Type
+
+from fastapi.datastructures import Default
 from fastapi.encoders import jsonable_encoder
 from fastapi.exceptions import RequestValidationError
 from starlette.exceptions import HTTPException
 from starlette.requests import Request
-from starlette.responses import JSONResponse
+from starlette.responses import JSONResponse, Response
 from starlette.status import HTTP_422_UNPROCESSABLE_ENTITY
 
 
-async def http_exception_handler(request: Request, exc: HTTPException) -> JSONResponse:
+async def http_exception_handler(
+    request: Request, exc: HTTPException,
+    *,
+    response_class: Type[Response] = Default(JSONResponse)
+) -> Type[Response]:
     headers = getattr(exc, "headers", None)
     if headers:
-        return JSONResponse(
+        return response_class(
             {"detail": exc.detail}, status_code=exc.status_code, headers=headers
         )
     else:
```

Comment:
Remove else statement here. No need that.

Reply:
👌